src/snapshot_middleware/project.rs

```diff
@@ -78,7 +80,42 @@ pub fn snapshot_project_node(
         // If the path specified in the project is relative, we assume it's
         // relative to the folder that the project is in, project_folder.
         let path = if path.is_relative() {
-            Cow::Owned(project_folder.join(path))
+            // Convert paths to use forward slashes for compatibility with the relative-path crate
+            let project_folder_with_slash_separator = match project_folder.to_slash() {
+                Some(p) => p,
+                None => return Ok(None),
```

Comment:
I don't think these returns are correct, it looks like we'll end up returning early in the case that we happen to be unable to convert paths. Do we have a good understanding of when this will happen?

Reply:
Should be obviated by rewrite to custom function